tools/cli/lib/api.c

```diff
@@ -964,6 +1023,10 @@ TDNFCliCheckUpdateCommand(
     {
         TDNFFreePackageInfoArray(pPkgInfo, dwCount);
     }
+    /* yum and dnf return 100 if there are package updates available.
+       puppet depends on this behaviour, even with tdnf. */
+    if(nCheckUpdateCompat && dwCount > 0)
```

Comment:
Please check for `dwError == 0`. We don't want to overwrite the error code with 100.